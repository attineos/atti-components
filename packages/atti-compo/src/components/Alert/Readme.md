Alert with error :

```js
<Alert variance="error">There is an error</Alert>
```

Alert with success :

```js
<Alert variance="success">Success</Alert>
```

Alert with information :

```js
<Alert variance="info">Information</Alert>
```

Alert with warning :

```js
<Alert variance="warning">Warning</Alert>
```
