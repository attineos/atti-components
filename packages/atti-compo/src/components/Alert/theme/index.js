// Create the theme part for Alert. Please add a call into the componentsFactory.
export default ({ border, colors, spaces, fonts }) => ({
  colors: {
    background: {
      error: colors.lightRed,
      success: colors.lightGreen,
      warning: colors.lightOrange,
      info: colors.lightBlue,
    },
    text: {
      error: colors.darkRed,
      success: colors.darkGreen,
      warning: colors.darkOrange,
      info: colors.darkBlue,
    },
    border: {
      error: colors.red,
      success: colors.green,
      warning: colors.orange,
      info: colors.blue,
    },
  },
  spaces: {
    padding: spaces.s6,
  },
  border: {
    radius: border.radius.br3,
    size: border.widths.bw1,
  },
  sizes: {
    width: '90%',
  },
  fonts: {
    fontWeight: fonts.fontWeight.fw800,
    lineHeight: fonts.lineHeight.lh10,
  },
})
