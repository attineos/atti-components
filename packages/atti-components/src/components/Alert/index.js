import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import StyledDiv from './styles/StyledDiv'

/**
 * An alert component.
 *
 */
const Alert = ({ children, variant, ...rest }) => (
  <StyledDiv variant={variant} {...rest}>
    {children}
  </StyledDiv>
)

Alert.defaultProps = {
  variant: 'success',
}

Alert.propTypes = {
  /**
   * Content of the Alert.
   */
  children: PropTypes.node.isRequired,
  /**
   * Type of alert.
   */
  variant: PropTypes.oneOf(['success', 'error', 'info', 'warning']),
}

/** @component */
export default styled(Alert)``
