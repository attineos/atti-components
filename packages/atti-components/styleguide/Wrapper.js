import React from 'react'

import ThemeProvider from '../src/components/ThemeProvider'
import { defaultTheme } from '../src/themes'

export default ({ children }) => <ThemeProvider otherTheme={defaultTheme}>{children}</ThemeProvider>
