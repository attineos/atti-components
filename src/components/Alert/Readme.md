Alert with error :

```js
<Alert type="error">There is an error</Alert>
```

Alert with success :

```js
<Alert type="success">Success</Alert>
```

Alert with information :

```js
<Alert type="info">Information</Alert>
```

Alert with warning :

```js
<Alert type="warning">Warning</Alert>
```
