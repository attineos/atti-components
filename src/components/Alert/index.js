import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import StyledDiv from './styles/StyledDiv'

/**
 * An alert component.
 *
 */
const Alert = ({ children, className, type }) => (
  <StyledDiv className={className} type={type}>
    {children}
  </StyledDiv>
)

Alert.defaultProps = {
  className: '',
  type: 'success',
}

Alert.propTypes = {
  /**
   * Content of the Alert.
   */
  children: PropTypes.node.isRequired,
  /**
   * Classes of the Alert.
   */
  className: PropTypes.string,
  /**
   * Type of alert.
   */
  type: PropTypes.oneOf(['success', 'danger', 'info', 'warning']),
}

/** @component */
export default styled(React.memo(Alert))``
