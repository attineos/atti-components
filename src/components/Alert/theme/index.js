// Create the theme part for Alert. Please add a call into the componentsFactory.
export default ({ borders, colors, spaces, fonts }) => ({
  colors: {
    background: {
      error: colors.supportLightDanger,
      success: colors.supportLightSuccess,
      warning: colors.supportLightWarning,
      info: colors.supportLightInfo,
    },
    text: {
      error: colors.supportDarkDanger,
      success: colors.supportDarkSuccess,
      warning: colors.supportDarkWarning,
      info: colors.supportDarkInfo,
    },
    borders: {
      error: colors.supportDanger,
      success: colors.supportSuccess,
      warning: colors.supportWarning,
      info: colors.supportInfo,
    },
  },
  spaces: {
    padding: spaces.s0_5,
  },
  borders: {
    radius: borders.radius.medium,
    size: borders.widths.bw2,
  },
  sizes: {
    width: 'auto',
  },
  fonts: {
    fontWeight: fonts.fontWeight.fw500,
    lineHeight: fonts.lineHeight.lh10,
  },
})
