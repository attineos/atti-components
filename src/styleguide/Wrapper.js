import React from 'react'
import PropTypes from 'prop-types'
import { remove, get } from 'lodash'

import Cookies from './helpers/Cookies'
import themes from './config/themes'

import ThemeProvider from '../components/ThemeProvider'

const updateWrappers = []

export const getWrappersToUpdate = () => updateWrappers

class Wrapper extends React.Component {
  static propTypes = {
    /**
     * The child node.
     */
    children: PropTypes.node.isRequired,
  }

  state = {
    selectedTheme: null,
  }

  componentWillMount() {
    updateWrappers.push(this)

    const selectedTheme = Cookies.hasItem('selectedTheme')
      ? Cookies.getItem('selectedTheme')
      : 'default'

    this.state.selectedTheme = get(themes, selectedTheme, null)
  }

  componentWillUnmount() {
    remove(updateWrappers, item => item === this)
  }

  handleSetTheme = newTheme =>
    this.setState(
      {
        selectedTheme: newTheme,
      },
      () => this.forceUpdate(),
    )

  render() {
    const { children } = this.props
    const { selectedTheme } = this.state

    return (
      <ThemeProvider theme={selectedTheme} key={Math.random()}>
        {children}
      </ThemeProvider>
    )
  }
}

export default Wrapper
