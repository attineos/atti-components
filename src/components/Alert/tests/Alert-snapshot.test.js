/* This file was generated automatically, please do not write in it. */
import React from 'react';
import Alert from '..';
import Wrapper from 'styleguide/Wrapper'
import renderer from 'react-test-renderer';

const mock = () => {}

it('Alert renders correctly', () => {
  const tree = renderer
    .create(
      <Wrapper>
        <Alert className="Atticomponents" variance={null} ><div/></Alert>
      </Wrapper>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
    