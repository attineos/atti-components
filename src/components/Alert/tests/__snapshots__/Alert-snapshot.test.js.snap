// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
<div
  className="StyledDiv-rsh67c-0 gyTSNi"
  type={null}
>
  <div />
</div>
`;
